test/integration/stats_integration_test.cc

```diff
@@ -286,7 +287,7 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithFakeSymbolTable) {
   // If you encounter a failure here, please see
   // https://github.com/envoyproxy/envoy/blob/master/source/docs/stats.md#stats-memory-tests
   // for details on how to fix.
-  EXPECT_MEMORY_EQ(m_per_cluster, 44233);
+  EXPECT_MEMORY_EQ(m_per_cluster, 44425);
```

Comment:
qq: Why does the memory change here? I would expect no change?

Reply:
I assumed it was because each cluster has its own resource manager and I'm fiddling with the resource type?

Reply:
But what changes are you making that cause it to grow in size? Is it padding or something? cc @jmarantz

Reply:
I think it's because `ManagedResourceImpl` has a concrete base class. All of that functionality used to be in the same class, and there's probably a difference in padding due to different ordering of the members between the two classes, or something like that.

Reply:
OK that's probably fine, it just seemed like a pretty large bump for no functionality change.